WooCommerce/Classes/Extensions/UIAlertController+Helpers.swift

```diff
@@ -58,6 +58,43 @@ extension UIAlertController {
         viewController.present(actionSheet, animated: true)
     }
 
+    /// Present an alert when the app does not have permission to use camera for barcode scanner.
+    /// The alert has an action that links to device settings and a cancel action.
+    static func presentBarcodeScannerNoCameraPermissionAlert(viewController: UIViewController,
+                                                             onCancel: (() -> Void)? = nil) {
+        presentAlertWithLinkToOpenSettings(viewController: viewController,
+                                           title: BarcodeScannerNoCameraPermissionAlert.Localization.title,
+                                           message: BarcodeScannerNoCameraPermissionAlert.Localization.message,
+                                           onCancel: onCancel)
+    }
+
+    /// Present an alert with an action that links to device settings and cancel action.
+    static func presentAlertWithLinkToOpenSettings(viewController: UIViewController,
+                                                   title: String? = nil,
+                                                   message: String? = nil,
+                                                   onCancel: (() -> Void)? = nil) {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        alert.view.tintColor = .text
+
+        let openSettingsAction = UIAlertAction(title: AlertWithLinkToOpenSettings.Localization.openSettings, style: .default) { _ in
+            AlertWithLinkToOpenSettings.openSettings()
+        }
+        alert.addAction(openSettingsAction)
+        alert.preferredAction = openSettingsAction
+
+        alert.addCancelActionWithTitle(AlertWithLinkToOpenSettings.Localization.cancel) { _ in
+            onCancel?()
+        }
+
+        if let popoverController = alert.popoverPresentationController {
+            popoverController.sourceView = viewController.view
+            popoverController.sourceRect = viewController.view.bounds
+            popoverController.permittedArrowDirections = []
+        }
```

Comment:
I'm not sure how this works, because from my testing on an iPad, the alert was not displayed as a popover:
![IMG_0066](https://user-images.githubusercontent.com/5533851/146295298-32fe71c0-4f69-4293-8006-ad7d74e34d99.PNG)

Plus, I think it'd be nicer if the popover is pointed from the source view (in this case the barcode button). Should we send the source view to the method to display it from there?

Reply:
From my testing, it's not possible to present as a popover with `.alert` style - `popoverPresentationController` is always `nil` where it is non-nil for `.actionSheet` style 😞 I couldn't find official documentation, but the alerts I saw on iPhone are also presented as an alert on iPad.

I removed the popover setup to avoid confusion in: https://github.com/woocommerce/woocommerce-ios/pull/5679/commits/0454fba3c813ed84dca148d2581d420176f1da99